src/TypeScript/EslintBridgeClient/EslintBridgeProcess.cs

```diff
@@ -158,8 +158,18 @@ private void StartServer()
 
         private void Process_Exited(object sender, EventArgs e)
         {
-            logger.LogDebug($"[eslint-bridge] [process id: {processId}] Process exited.", processId);
-            ClearProcessData();
+            // Guard against a potential race condition i.e. this event being received
```

Comment:
Simple fix for the race condition you identified. There may still be other issues, but this is an improvement.